EIPS/eip-4410.md

```diff
@@ -0,0 +1,160 @@
+    /// @param _tokenId - the NFT asset that the memory is added to 
+    /// @param metadataURI - the URI that points to the memory metadata
+    function addMemory(uint256 _tokenId, string calldata _metadataURI) external {
+        require(_exists(_tokenId), "ERC721: operator query for nonexistent token");
+        require(_isApprovedOrOwner(_msgSender(), _itemId), "CONTRACT::addMemory: caller is not owner nor approved");
+        emit LogMemoryEvent(_itemId, _metadataURI);
+    }
+
+    event LogMemoryEvent(uint256 indexed itemId, string metadataURI );
+}
+
+interface IERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+```
+
+Implementers of this standard **MUST** implement the metadata using the following JSON schema (note that only tokenId and externalUrl are required):
+
+```yaml
+type: object
+properties:
+  tokenId:
+    type: number
+  externalUrl:
+    type: string
+  files:
+    type: array
+    items:
+      type: object
+      properties:
+        type:
+          type: string
+        uri:
+          type: string
+  notes:
+    type: string
+  geoInfo:
+    type: object
+    properties:
+      latitude:
+        type: number
+      longitude:
+        type: number
+  country_name:
+    type: string
+  region_name:
+    type: string
+  time_zone:
+    type: string
+  zip_code:
+    type: string
+  region_code:
+    type: string
+  ip:
+    type: string
+  userAgent:
+    type: string
+  userId:
+    type: string
+required: [
+    "tokenId", "externalUrl"
+  ]
+```
+
+Here is an example of this JSON schema.
+
+```json
+{
+    "tokenId":410,
+    "externalUrl":"https://polygonstaging.realitems.io/items/Oo4oGL4Y",
+    "files":[ 
+        { 
+            "type": "image", 
+            "uri": "ipfs://QmfJQyoTQBDPovYdrJgReFXarrzvP6G67MDxBESizredM" 
+        },
+        {
+            "type": "video",
+            "uri": "https://ipfs.io/ipfs/QmXYyFCE3fHaGi4QxqqamPj5JV3sqtgwByqsAsHYgib77"
+        },
+        {
+            "type": "sound",
+            "uri": "https://ipfs.io/ipfs/QmXYyFCE3fHaGi4QxqqamPj5JV3sqtgwByqsACsHYib77"
+        }
+    ],
+    "notes":"This is a memory",
+    "geoInfo": {
+        "latitude":37.976,
+        "longitude":-122.3359
+    },
+    "country_name":"United States",
+    "region_name":"California",
+    "time_zone":"America/Los_Angeles",
+    "zip_code":"94806",
+    "region_code":"CA",
+    "ip":"43.199.84.252",
+    "userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36",
+    "userId":"ken@realitems.org"
+}
+```
```

Comment:
There are a few sections missing here - check [EIP-1](https://eips.ethereum.org/EIPS/eip-1#what-belongs-in-a-successful-eip):
* Rationale
* Backwards Compatability
* Test Cases
* Reference Implementation (optional, but imho very good to have in ERCs)
* Security Considerations
* Copyright Waiver

Reply:
bump